core/services/workflows/syncer/engine_registry.go

```diff
@@ -36,6 +36,26 @@ func (r *engineRegistry) Get(id string) (*workflows.Engine, error) {
 	return engine, nil
 }
 
+// IsRunning is true if the engine exists and is ready.
+func (r *engineRegistry) IsRunning(id string) bool {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
+	engine, found := r.engines[id]
+	if !found {
+		return false
+	}
+
+	return engine.Ready() == nil
+}
+
+// Exists checks if an engine exists in the registry.
+func (r *engineRegistry) Exists(id string) bool {
```

Comment:
nit/ something like `Has` or `Contains` might be more idiomatic here
```suggestion
// Has checks if an engine exists in the registry.
func (r *engineRegistry) Has(id string) bool {
```

Reply:
e.g. as with https://pkg.go.dev/slices#Contains

Reply:
- https://pkg.go.dev/strings#Contains
- https://pkg.go.dev/strings#HasPrefix

Reply:
unused, removed.